Tests/NIOFileSystemIntegrationTests/FileSystemTests.swift

```diff
@@ -1681,6 +1681,22 @@ extension FileSystemTests {
         XCTAssertGreaterThan(info.size, 0)
     }
 
+    func testTemporaryDirectoryRespectsEnvironmentVariable() async throws {
+        let originalTMPDIR = getenv("TMPDIR")
+        defer {
+            if let originalTMPDIR {
+                setenv("TMPDIR", String(cString: originalTMPDIR), 1)
+            } else {
+                unsetenv("TMPDIR")
+            }
+        }
+
+        let customTmpDir = "/custom/tmp/dir"
+        setenv("TMPDIR", customTmpDir, 1)
```

Comment:
I'm a bit hesitant to mess around with `TMPDIR` like this in case other tests running in parallel are relying on it. 

II think my preference here would be to check if `TMPDIR` is set and then check `fs.temporaryDirectory` if it was set.

Reply:
My bad, I haven't considered tests running in parallel, I see how this is potentially problematic.